test/integration/consul-container/test/gateways/http_route_test.go

```diff
@@ -260,3 +232,176 @@ func TestHTTPRouteFlattening(t *testing.T) {
+	validRouteName := randomName("route", 16)
+	validPath := "/foo"
+	invalidPath := "/bar"
+
+	//write config entries
+	proxyDefaults := &api.ProxyConfigEntry{
+		Kind:      api.ProxyDefaults,
+		Name:      api.ProxyConfigGlobal,
+		Namespace: namespace,
+		Config: map[string]interface{}{
+			"protocol": "http",
+		},
+	}
+
+	_, _, err := client.ConfigEntries().Set(proxyDefaults, nil)
+	assert.NoError(t, err)
+
+	apiGateway := createGateway(gatewayName, "http", listenerPort)
+
+	invalidRoute := &api.HTTPRouteConfigEntry{
+		Kind: api.HTTPRoute,
+		Name: invalidRouteName,
+		Parents: []api.ResourceReference{
+			{
+				Kind:      api.APIGateway,
+				Name:      gatewayName,
+				Namespace: namespace,
+			},
+		},
+		Hostnames: []string{
+			"test.foo",
+		},
+		Namespace: namespace,
+		Rules: []api.HTTPRouteRule{
+			{
+				Filters: api.HTTPFilters{
+					URLRewrite: &api.URLRewrite{
+						Path: "/v1/invalid",
+					},
+				},
+				Services: []api.HTTPService{
+					{
+						Name:      invalidService.GetServiceName(),
+						Namespace: namespace,
+					},
+				},
+				Matches: []api.HTTPMatch{
+					{
+						Path: api.HTTPPathMatch{
+							Match: api.HTTPPathMatchPrefix,
+							Value: "/bar",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	validRoute := &api.HTTPRouteConfigEntry{
+		Kind: api.HTTPRoute,
+		Name: validRouteName,
+		Parents: []api.ResourceReference{
+			{
+				Kind:      api.APIGateway,
+				Name:      gatewayName,
+				Namespace: namespace,
+			},
+		},
+		Hostnames: []string{
+			"test.foo",
+		},
+		Namespace: namespace,
+		Rules: []api.HTTPRouteRule{
+			{
+				Filters: api.HTTPFilters{
+					URLRewrite: &api.URLRewrite{
+						Path: "/v1/api",
+					},
+				},
+				Services: []api.HTTPService{
+					{
+						Name:      validService.GetServiceName(),
+						Namespace: namespace,
+					},
+				},
+				Matches: []api.HTTPMatch{
+					{
+						Path: api.HTTPPathMatch{
+							Match: api.HTTPPathMatchPrefix,
+							Value: "/foo",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, _, err = client.ConfigEntries().Set(apiGateway, nil)
+	assert.NoError(t, err)
+	_, _, err = client.ConfigEntries().Set(invalidRoute, nil)
+	assert.NoError(t, err)
+	_, _, err = client.ConfigEntries().Set(validRoute, nil)
+	assert.NoError(t, err)
+
+	//create gateway service
+	gatewayService, err := libservice.NewGatewayService(context.Background(), gatewayName, "api", cluster.Agents[0], listenerPort)
+	require.NoError(t, err)
+	libassert.CatalogServiceExists(t, client, gatewayName)
+
+	//make sure config entries have been properly created
+	checkGatewayConfigEntry(t, client, gatewayName, namespace)
+	checkHTTPRouteConfigEntry(t, client, invalidRouteName, namespace)
+	checkHTTPRouteConfigEntry(t, client, validRouteName, namespace)
+
+	ip := "localhost"
+	gatewayPort, err := gatewayService.GetPort(listenerPort)
+	assert.NoError(t, err)
+
+	//TODO these were the assertions we had in the original test, but I'm unclear on why invalid is invalid
```

Comment:
Yeah, not quite sure what's going on here, but I'm assuming this test is just supposed to show two different path-prefix values being stripped? Maybe rename the services from valid/invalid?